File: eAutoShop/eAutoShop.Services/Database/StaffReview.cs
using System;
using System.Collections.Generic;

namespace eAutoShop.Services.Database;

public partial class StaffReview
{
    public int Id { get; set; }

    public int? UserId { get; set; }

    public int? EmployeeId { get; set; }

    public int? AppointmentId { get; set; }

    public int? Rating { get; set; }

    public string? Comment { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual Appointment? Appointment { get; set; }

    public virtual User? Employee { get; set; }

    public virtual User? User { get; set; }
}

File: eAutoShop/eAutoShop.UI/eautoshop_mobile/lib/models/staff_review/staff_review.dart
import 'package:json_annotation/json_annotation.dart';

part 'staff_review.g.dart';

@JsonSerializable()
class StaffReview {
  int id;
  int? userId;
  int? employeeId;
  int? appointmentId;
  int? rating;
  String? comment;
  DateTime? createdAt;
  String? userName;
  String? employeeName;

  StaffReview(
    this.id,
    this.userId,
    this.employeeId,
    this.appointmentId,
    this.rating,
    this.comment,
    this.createdAt,
    this.userName,
    this.employeeName,
  );

  factory StaffReview.fromJson(Map<String, dynamic> json) =>
      _$StaffReviewFromJson(json);

  Map<String, dynamic> toJson() => _$StaffReviewToJson(this);
}
